armada_jupyter/armada_utils.py

```diff
@@ -32,27 +33,32 @@ def create_armada_request(
     )
 
 
-def submit(submission: Submission, job: Job, client: ArmadaClient) -> str:
+def submit(submission: Submission, job: Job, client: ArmadaClient) -> Tuple[str, str]:
     """
     Submits a workflow for jupyter notebooks.
     """
 
     queue = submission.queue
     job_set_id = submission.job_set_id
 
+    # Inject the URL for the Jupyter Notebook
+    job, url_start, url_end = inject_url(job)
+
     # Create the PodSpec for the job
     job_request_items = [create_armada_request(job, client)]
     resp = client.submit_jobs(
         queue=queue, job_set_id=job_set_id, job_request_items=job_request_items
     )
 
     job_id = resp.job_response_items[0].job_id
-    return job_id
+    url = f"{url_start}{job_id}{url_end}"
+    return job_id, url
 
 
-def construct_url(job: Job, job_id: str) -> str:
+def inject_url(job: Job) -> Tuple[Job, str, str]:
```

Comment:
This seems more complicated than it needs to be.  

Why don't you just build the url and return the url as a single string?

Reply:
Basically, I need to know the url parts before I submit the job (i.e I don't know the job-id yet), but I still want to print the full url in terminal.

This was my best attempt at following DRY